src/scmrepo/git/lfs/fetch.py

```diff
@@ -121,16 +143,13 @@ def _collect_objects(
 def _filter_paths(
     paths: Iterable[str], include: Optional[list[str]], exclude: Optional[list[str]]
 ) -> Iterator[str]:
-    filtered = set()
     if include:
-        for pattern in include:
-            filtered.update(fnmatch.filter(paths, pattern))
-    else:
-        filtered.update(paths)
+        include_match = re.compile(r"|".join(map(fnmatch.translate, include))).match
```

Comment:
let's do this a separate PR please? also, if we have a lot of patterns (100s) - will it handle it fine?

Reply:
Agreed, I've dropped the commit here and created submitted a separate PR at #356.

The unionized regex can easily handle 100s – even orders of magnitude more – patterns without any problems. Check this example:

```python
import fnmatch
import re

re.compile(r"|".join(map(fnmatch.translate, [f"/a/b/{i}/**" for i in range(100000)]))).match("/a/b/1/")
# <re.Match object; span=(0, 7), match='/a/b/1/'>
```

Also note that unionizing the derived regexes is an [undocumented but known technique](https://github.com/python/cpython/blob/11594da0461b02a8448dfa3354884ef6e274e587/Lib/fnmatch.py#L163-L165).

Reply:
yep, in this case I'm still a bit worried bc it could explode (we don't really control the number of files) - also, I don't know about the performance implications (how much time compile takes, if we run each file by this regex - is it really better vs running one by one). Anyways, we can continue the discussion in the new PR. Thanks for addressing this.!